scripts/west_commands/packages.py

```diff
@@ -0,0 +1,149 @@
+            "If this option is not given, the 'packages' command will run for Zephyr "
+            "and all active projects.",
+        )
+
+        subparsers_gen = parser.add_subparsers(
+            metavar="<manager>",
+            dest="manager",
+            help="select a manager.",
+            required=True,
+        )
+
+        pip_parser = subparsers_gen.add_parser(
+            "pip",
+            help="manage pip packages",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Manage pip packages:
+
+                Run 'west packages pip' to print all pip packages or requirements
+                files needed by Zephyr and the active west projects.
+
+                The output is compatible with the requirements file format itself.
+            """
+            ),
+        )
+
+        pip_parser.add_argument(
+            "--install",
+            action="store_true",
+            help="Install pip packages instead of listing them. "
+            "A single 'pip install' command is built and executed. "
+            "Additional pip arguments can be passed after a -- separator "
+            "from the original 'west packages pip --install' command. For example pass "
+            "'--dry-run' to pip not to actually install anything, but print what would be.",
+        )
+
+        return parser
+
+    def do_run(self, args, unknown):
+        if len(unknown) > 0 and unknown[0] != "--":
+            self.die(
+                f'Unknown argument "{unknown[0]}"; '
+                'arguments for the manager should be passed after "--"'
+            )
+
+        if args.manager == "pip":
+            return self.do_run_pip(args, unknown[1:])
+
+        # Unreachable but print an error message if an implementation is missing.
+        self.die(f'Unsupported package manager: "{args.manager}"')
+
+    def do_run_pip(self, args, manager_args):
+        packages = []
+        requirements = []
+
+        if not args.projects or "zephyr" in args.projects:
+            requirements.append(ZEPHYR_BASE / "scripts/requirements.txt")
+
+        for module in zephyr_module.parse_modules(ZEPHYR_BASE, self.manifest):
+            module_name = module.meta.get("name", None)
+            if args.projects and module_name not in args.projects:
+                continue
+
+            # Get the optional pip section from the package managers
+            pip = module.meta.get("package-managers", {}).get("pip")
+            if pip is None:
+                continue
+
+            # Add individual listed packages
+            packages += pip.get("packages", [])
+            # Add requirements files
+            requirements += [Path(module.project) / r for r in pip.get("requirements", [])]
+
+        if args.install:
+            if not in_venv():
+                self.wrn("Running pip install outside of a virtual environment")
```

Comment:
I would argue that this should be an error unless overridden because like how you get domain squatters pretending to be other things, you could easily get something pretending (and looking just like zephyr) with a link to official documentation on using this and it has a module with something dodgy in it, as to what command line argument I don't know, ``allow-dangerous-commands``? Doesn't really sound good but I would err on the side of caution so that users can see the command should not just be used randomly if someone tells them to

Reply:
Updated